huaweicloud/services/css/resource_huaweicloud_css_thesaurus.go

```diff
@@ -0,0 +1,225 @@
+package css
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/chnsz/golangsdk"
+	"github.com/chnsz/golangsdk/openstack/css/v1/thesaurus"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+)
+
+func ResourceCssthesaurus() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: ResourceCssthesaurusCreate,
+		ReadContext:   ResourceCssthesaurusRead,
+		UpdateContext: ResourceCssthesaurusCreate,
+		DeleteContext: ResourceCssthesaurusDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"cluster_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"bucket_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"main_object": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"stop_object": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"synonym_object": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"update_time": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func ResourceCssthesaurusCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+	cssV1Client, err := config.CssV1Client(region)
+	if err != nil {
+		return diag.Errorf("Error creating CSS V1 client: %s", err)
+	}
+	opts := buildThesaurusCreateParameters(d)
+	clusterId := d.Get("cluster_id").(string)
+
+	loadErr := thesaurus.Load(cssV1Client, clusterId, *opts)
+	if loadErr.Err != nil {
+		return diag.Errorf("load thesaurus to css cluster failed. cluster_id=%s,error=%s", clusterId, loadErr.Err)
+	}
+
+	createResultErr := checkThesaurusLoadResult(ctx, cssV1Client, clusterId, d.Timeout(schema.TimeoutCreate))
+	if createResultErr != nil {
+		return diag.FromErr(createResultErr)
+	}
+
+	d.SetId(clusterId)
```

Comment:
move setId before waitfor, and terraform manages this resource

Reply:
if  check result is not success ,should not manage this resource

Reply:
If the resource has been created but check result is unsuccessful or the status is not expected, maybe the resource will always exist in HuaweiCloud. The customer does not know that an incorrect resource has been created, and it cannot be deleted through Terraform.

As long as a resource is created, it must be managed by Terraform, no matter what the state of the resource, otherwise it will produce unmanageable resources.

Reply:
If the resource has been created but check result is unsuccessful or the status is not expected, maybe the resource will always exist in HuaweiCloud. The customer does not know that an incorrect resource has been created, and it cannot be deleted through Terraform.

As long as a resource is created, it must be managed by Terraform, no matter what the state of the resource, otherwise it will produce unmanageable resources.